python/pyspark/sql/connect/client/core.py

```diff
@@ -296,7 +296,10 @@ def userAgent(self) -> str:
             or "_SPARK_CONNECT_PYTHON" when not specified.
             The returned value will be percent encoded.
         """
-        user_agent = self.params.get(ChannelBuilder.PARAM_USER_AGENT, "_SPARK_CONNECT_PYTHON")
+        user_agent = self.params.get(
+            ChannelBuilder.PARAM_USER_AGENT,
+            os.getenv("SPARK_CONNECT_USER_AGENT", "_SPARK_CONNECT_PYTHON"),
```

Comment:
The value of the env var needs to be appended to the existing user agent, not replaced.

Reply:
It gets prepended [here](https://github.com/apache/spark/blob/8d046de2cf2fe428c3eff1696f899eff30d0b4b7/python/pyspark/sql/connect/client/core.py#L290) to the rest of the current `userAgent`. The environment variable only gets used as a fallback option if the user did not specify their own user_agent in the provided `ChannelBuilder`.